docs/develop/topics/custom_views.md

```diff
@@ -0,0 +1,256 @@
+        {% endblock %}
+    </div>
+{% endblock %}
+```
+
+We added a `<div>` element (`root-container`) to later render a react application on it.
+
+## Register the view
+
+Now we can register the view we configured in `support.py`. To do this, we will open the `views.py` file in `./site/my_site` and add a new url rule within the `create_blueprint` function like the following:
+
+```python
+blueprint.add_url_rule(
+    "/support",
+    view_func=MySiteSupport.as_view("support_form"),
+)
+```
+
+Our full `views.py` file now looks like this:
+
+```python
+from flask import Blueprint
+from .support.support import MySiteSupport
+
+def create_blueprint(app):
+    blueprint = Blueprint(
+        "my_site",
+        __name__,
+        template_folder="./templates",
+    )
+
+    blueprint.add_url_rule(
+        "/support",
+        view_func=MySiteSupport.as_view("support_form"),
+    )
+
+    return blueprint
+```
+
+That is really all you need to get your custom view available on your desired path. If you now open your instance on `/support`, you will see the new template we added, `support.html`:
+
+![Custom view](./img/mysite_custom_views/mysite_custom_view.png)
+
+## Adding JavaScript to your template
+
+If you want, or need, to use JavaScript for your template, we will need to configure webpack for the site folder. This is done in the predefined `webpack.py` file of `site/my_site`.
+
+We can start by creating a js file in our assets folder:
+
+```
+├── site
+│   ├── my_site
+│   │   ├── assets
+│   │   │   ├── semantic-ui
+│   │   │   │   ├── js
+│   │   │   │   │   ├── my_site
+│   │   │   │   │   │   └── support.js
+│   │   ├── support
+│   │   ├── templates
+```
+
+Now we need to register this file in our webpack configuration, so that it will be available for us to use through our instance.
+
+In our `webpack.py` we will add a new entry point, pointing to the file we just created:
+
+```python
+from invenio_assets.webpack import WebpackThemeBundle
+
+theme = WebpackThemeBundle(
+    __name__,
+    "assets",
+    default="semantic-ui",
+    themes={
+        "semantic-ui": dict(
+            entry={
+                "my-site-support": "./js/my_site/support.js",
+            },
+        ),
+    },
+)
+```
+
+Here you can see that we made an alias for our js-file: `my-site-support`, which is the name we will use to reference the file.
+
+The JavaScript file could be anything that should be executed when the page is rendered, as an example it can render a simple Semantic UI Form:
+
+```javascript
+import React from 'react';
+import ReactDOM from "react-dom";
+import { Button, Form, Label } from 'semantic-ui-react'
+
+const rootContainer = document.getElementById("root-container");
+
+ReactDOM.render(
+    <Form>
+        <Form.Field>
+            <Label>First Name</Label>
+            <input placeholder='First Name' />
+        </Form.Field>
+        <Form.Field>
+            <Label>Last Name</Label>
```

Comment:
nit: I would add field called `name` and another textarea called `message`. Also, can you update the screenshot not to show the top banner of zenodo rdm?